### DISCUSSION TITLE
MFXTableView background color for odd/even rows

### DISCUSSION THREAD

**Discussion by julianprimetshofer** [palexdev/MaterialFX#236] [Archived]
Hey,

I just started working with MaterialFX and I'm currently trying to create a table and to make it easier for the users to find data in the table I want to change the background color of odd rows. Somehow I wasn't able to find anything concerning this but maybe I just didn't realize it. Only thing I found is that its possible to change it for the legacy table but I don't feel like using it because a lot of the awesome features of the new tableview are missing.

**Comment by palexdev** [comment]
The new table view doesn't implement odd/even rows but it's pretty simple to do so.
You just need to create a custom row and create the two needed PseudoClasses like this:
```java
PseudoClass EVEN_PSEUDO_CLASS = PseudoClass.getPseudoClass("even");
PseudoClass ODD_PSEUDO_CLASS = PseudoClass.getPseudoClass("odd");
```

Then you must update them every time the cell index changes, two ways: a listener on the index property or just do it in the updateIndex method (try this first, less memory required). Also keep in mind that when a cell in even the odd PseudoClass must be deactivated and the other way around

**Reply by julianprimetshofer** [reply]
Thanks for getting back to me. As you said I created a custom row class:
```java
public class MFXCustomTableRow extends MFXTableRow {

    PseudoClass EVEN_PSEUDO_CLASS = PseudoClass.getPseudoClass("even");
    PseudoClass ODD_PSEUDO_CLASS = PseudoClass.getPseudoClass("odd");

    public MFXCustomTableRow(MFXTableView tableView, Object data) {
        super(tableView, data);
    }

    @Override
    public void updateIndex(int index) {
        super.updateIndex(index);
        if(index % 2 != 0) {
            pseudoClassStateChanged(ODD_PSEUDO_CLASS, true);
            pseudoClassStateChanged(EVEN_PSEUDO_CLASS, false);
        } else {
            pseudoClassStateChanged(ODD_PSEUDO_CLASS, false);
            pseudoClassStateChanged(EVEN_PSEUDO_CLASS, true);
        }

    }
}
```

In my view class I'm constructing the tableview like this:
```java
MFXTableView<Result> resultTableView = new MFXTableView<>();
MFXTableColumn<Result> resultName = new MFXTableColumn<>("Result");
MFXTableColumn<Result> resultValue = new MFXTableColumn<>("Value");

resultName.setRowCellFactory(result -> new MFXTableRowCell<>(Result::getName));
resultValue.setRowCellFactory(result -> new MFXTableRowCell<>(Result::getResult));

resultTableView.getTableColumns().addAll(resultName, resultValue);
```

but how do I make use of my new "MFXCustomTableRow" class?

Sorry, but I have never worked with anything like this

**Reply by palexdev** [reply]
On the table view there's a method to set the row factory property
You got it right anyway, well done

**Reply by julianprimetshofer** [reply]
Lovely! Works perfectly fine.

Appreciate your help

**Reply by palexdev** [reply]
No problem

I'm still not sure about it but probably in the next version there will be a huge overhaul of controls using virtualized containers (lists, tables, combos...)
I'll keep this use case in mind eventually 👍🏻
